arena-core/src/main/java/org/uqbar/arena/widgets/Widget.java

```diff
@@ -22,15 +23,15 @@
  * @author npasserini
  */
 public class Widget implements Loggeable {
-	private Collection<Binding<?, Widget, WidgetBuilder>> bindings = CollectionFactory.createCollection();
-	protected List<Closure<WidgetBuilder>> configurations = CollectionFactory.createList();
+	private Collection<Binding<?, Widget, WidgetBuilder>> bindings = new ArrayList<>();
```

Comment:
Jajajaj, en Xtend ahora esa idea se podría hacer con extension methods.
Pero bueno, supongo que "ya fue".

2016-08-11 14:07 GMT+02:00 javierfernandes <notifications@github.com>:

> In arena-core/src/main/java/org/uqbar/arena/widgets/Widget.java
> <https://github.com/uqbar-project/arena/pull/8#discussion_r74409909>:
>
> > @@ -22,15 +23,15 @@
> >   * @author npasserini
> >   */
> >  public class Widget implements Loggeable {
> > -	private Collection<Binding<?, Widget, WidgetBuilder>> bindings = CollectionFactory.createCollection();
> > -	protected List<Closure<WidgetBuilder>> configurations = CollectionFactory.createList();
> > +	private Collection<Binding<?, Widget, WidgetBuilder>> bindings = new ArrayList<>();
>
> CollectionFactory era una idea de @npasserini
> <https://github.com/npasserini> que desarrollamos en otro "proyecto". La
> idea era introducir un nivel de abstracción al momento de decidir una
> implementación en particular de una colección.
> En un proyecto grande la mayoría de las veces el programador (me incluyo)
> instancia una colección "de memoria" sin pensar demasiado cual conviene
> para su uso. No hablo de elegir entre Set y List. Digo, ArrayList ? o
> LinkedList ?
> Salvo que uno esté ya codificando un algoritmo sensible o de bajo nivel,
> donde ya lo puede considerer en el mismo momento, en general uno está más
> ocupado de la funcionalidad que la performance.
> La performance aparece como un paso posterior.
>
> El punto es que al tener la indirección y con con algo de magia uno podría
> laburar en esas optimizaciones como "aspectos" sin tocar el código base.
> Onda, después describís en otro lado
>
> Widget.bindings = LinkedList
>
> Y CollectionFactory haría la magia de cuando se la pide en ese punto del
> código darle una instancia de LinkedList, Y desde otros puntos del código
> darle otras instancias de clases particulares.
>
> También es verdad el side effect the los generics.
> Si no recuerdo mal, las primeras versiones de generics de java no tenían
> esto
>
> new ArrayList<>
>
> Te forzaba a volver a escribir todo el tipo a la derecha.
>
> Bueno, eso, 2 años después, pero vino la explicación.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/uqbar-project/arena/pull/8/files/797af34680cac9217346bc335da039d79ab65424#r74409909>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEa1OZJU5cST6vHwRyOJWz75IlYl5aLRks5qexB4gaJpZM4C5ng4>
> .
>